client/data/agency-dashboard/use-fetch-dashboard-sites.ts

```diff
@@ -61,12 +62,18 @@ const useFetchDashboardSites = (
 				totalFavorites: data.total_favorites,
 			};
 		},
-		onError: () =>
-			dispatch(
-				errorNotice( translate( 'Failed to retrieve your sites. Please try again later.' ) )
-			),
 		enabled: isPartnerOAuthTokenLoaded,
 	} );
+
+	useEffect( () => {
+		if ( data.isError ) {
+			dispatch(
+				errorNotice( translate( 'Failed to retrieve your sites. Please try again later.' ) )
```

Comment:
I think it's good pratice to add an `id` and `duration` option to usages of `errorNotice` and would suggest in this case, too. Like this:

```suggestion
				errorNotice(
				  translate( 'Failed to retrieve your sites. Please try again later.' ),
				  { id: 'dashboard-sites-fetch-failure', duration: 5 * 1000 }
				)
```

Reply:
Ah, something to add here. I'd consider moving this effect to the consumer component of `useFetchDashboardSites`. We've been trying to have those error dispatches close to the relevant UI code.

Note: there's some good suggestions on where to put stuff in this comment: https://github.com/TanStack/query/discussions/5279#discussioncomment-5632938 as well which I think make sense for our codebase, too.